massa-execution-worker/src/slot_sequencer.rs

```diff
@@ -209,9 +209,16 @@ impl SlotSequencer {
     pub fn update(
         &mut self,
         mut new_consensus_final_blocks: HashMap<Slot, BlockId>,
+        #[cfg_attr(feature = "slot-replayer", allow(unused_assignments))]
         mut new_blockclique: Option<HashMap<Slot, BlockId>>,
         mut new_blocks_metadata: PreHashMap<BlockId, ExecutionBlockMetadata>,
     ) {
+        // as a precaution, if we don't want to process speculative slots, clear the speculative feed just in case
```

Comment:
Maybe add a note to specify that in the slot replayer we are only going to process final slot ?

Reply:
done